### PR TITLE
Registration Form

### PR DESCRIPTION
**Pull request by bbengfort**
Re-implements the registration form with basic functionality and the ability to download the JSON data required to make a registration. Intended as a temporary hold over for copy review until we can hook up the backend.

Also implements a docker-compose functionality in order to develop against the front-end and alerts handling for errors when using the directory service.

### CODE REVIEW

File: web/src/components/Registration.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                      To assist in completing the registration form, which is somewhat lengthy, the
+                      form is broken into multiple sections, with information stored in your <em>local
+                      browser storage</em> so that you can come back and complete the process. <strong>No
+                      information is sent until you submit the form in the summary section</strong>.
+                    </p>
+                    <p className="text-danger">
+                      The registration form is currently in its beta implementation. You will not be
+                      able to submit the form now, but can download the form to submit it via the CLI
+                      registration interface. The registration form will be completed when the TRISA
+                      v1beta1 protocol is released.
```
Comment (rebeccabilbro) [inline web/src/components/Registration.js:114]: Suggest updating the copy to read:

This registration form is currently in its beta implementation. You will not be able to submit the form now, but can download it and submit via the CLI registration interface. This registration form will become live when the TRISA v1beta1 protocol is released.

File: web/src/components/Registration.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+              <Tab.Content>
+                <Tab.Pane eventKey="introduction">
+                  <fieldset>
+                    <legend>Introduction</legend>
+                    <p>
+                      Thank you for your interest in the TRISA network for Travel Rule Compliance.
+                      This multi-part form is the first step in the registration and certificate
+                      issuance process. The information you provide will be used to verify the
+                      legal entity that you represent and, where appropriate, will be available to
+                      verified TRISA members to facilitate compliance-decisions.
```
Comment (rebeccabilbro) [inline web/src/components/Registration.js:102]: compliance decisions (no hyphen)

File: web/src/components/Registration.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+                  </fieldset>
+                </Tab.Pane>
+                <Tab.Pane eventKey="trixo">
+                  <fieldset>
+                    <legend>TRIXO Questionnaire</legend>
+                    <p>
+                      This questionnaire is designed to help the TRISA working group and other
+                      TRISA members understand the regulatory regime of your organization to ensure
+                      that required compliance information exchanges are conducted correctly and
+                      safely. All verified TRISA members will have access to this information.
```
Comment (rebeccabilbro) [inline web/src/components/Registration.js:236]: Recommend breaking the text up into shorter sentences to support translation:

This questionnaire is designed to help the TRISA working group and TRISA members understand the regulatory regime of your organization. The information you provide will help ensure that required compliance information exchanges are conducted correctly and safely. All verified TRISA members will have access to this information.

File: web/src/components/TRIXO.js [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+          </Form.Text>
+        </Form.Group>
+      </fieldset>
+      <fieldset>
+        <legend className="sublegend">Data Protection Policies</legend>
+        <Form.Group>
+          <Form.Label>Is your organization required by law to safeguard PII?</Form.Label>
+          <Form.Check
+            type="switch"
+            id="mustSafegaurdPII"
```
Comment (rebeccabilbro) [inline web/src/components/TRIXO.js:247]: Safegaurd => Safeguard

File: web/src/components/TRIXO.js [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+            label="Must Safegaurd PII"
```
Comment (rebeccabilbro) [inline web/src/components/TRIXO.js:248]: Safegaurd => Safeguard

File: web/src/components/TRIXO.js [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+            checked={data.must_safeguard_pii}
+            onChange={createBoolChangeHandler("must_safeguard_pii")}
+          />
+        </Form.Group>
+        <Form.Group>
+          <Form.Label>Does your organization secure and protect PII, including PII received from other VASPs under the Travel Rule?</Form.Label>
+          <Form.Check
+            type="switch"
+            id="safegaurdsPII"
```
Comment (rebeccabilbro) [inline web/src/components/TRIXO.js:257]: safegaurdsPII => safeguardsPII

File: web/src/components/TRIXO.js [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+            label="Safegaurds PII"
```
Comment (rebeccabilbro) [inline web/src/components/TRIXO.js:258]: Safegaurds PII => Safeguards PII

File: web/src/components/TRIXO.js [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+            checked={data.safegaurds_pii}
```
Comment (rebeccabilbro) [inline web/src/components/TRIXO.js:259]: data.safegaurds_pii => data.safeguards_pii

File: web/src/components/TRIXO.js [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+            onChange={createBoolChangeHandler("safegaurds_pii")}
```
Comment (rebeccabilbro) [inline web/src/components/TRIXO.js:260]: safegaurds_pii => safeguards_pii

File: web/src/components/Registration.js [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+        has_required_regulatory_program: "",
+        conducts_customer_kyc: false,
+        kyc_threshold: 0.0,
+        kyc_threshold_currency: "USD",
+        must_comply_travel_rule: false,
+        applicable_regulations: ["FATF Recommendation 16"],
+        compliance_threshold: 0.0,
+        compliance_threshold_currency: "USD",
+        must_safeguard_pii: false,
+        safegaurds_pii: false,
```
Comment (rebeccabilbro) [inline web/src/components/Registration.js:46]: safegaurds_pii => safeguards_pii

**Review by rebeccabilbro — APPROVED** [review]
Looking great! Just a few small typos and suggestions on the copy.

File: web/src/components/Registration.js (hunk 3 above)
Comment (bbengfort) [reply web/src/components/Registration.js:384]: Great point!

File: web/src/components/TRIXO.js (hunk 4 above)
Comment (bbengfort) [reply web/src/components/TRIXO.js:247]: I will check to ensure that it's also spelled correctly in the protocol buffers!

**Pull request merged**